heron/shell/src/python/utils.py

```diff
@@ -135,7 +135,7 @@ def read_chunk(filename, offset=-1, length=-1, escape_data=False):
   if length == -1:
     length = fstat.st_size - offset
 
-  with open(filename, "r") as fp:
```

Comment:
I think the `encoding='utf-8'` is implied on normal builds, including WIndows as of python3.6, so the change on this line could be dropped

Reply:
to be consistent with the previous behaviour (not sure it still makes sense), I think this would be something like:
```python
with open(filename, "rb") as fp:
   ...
   if data:
       # use permissive decoding and escaping if escape_data is set, otherwise use strict decoding
       data = _escape_data(data) if escape_data else data.decode()